src/ucp/api/ucp_def.h

```diff
@@ -208,8 +214,9 @@ typedef struct ucp_mem_attr {
  * present. It is used to enable backward compatibility support.
  */
 enum ucp_mem_attr_field {
-    UCP_MEM_ATTR_FIELD_ADDRESS = UCS_BIT(0), /**< Virtual address */
-    UCP_MEM_ATTR_FIELD_LENGTH  = UCS_BIT(1)  /**< The size of memory region */
+    UCP_MEM_ATTR_FIELD_ADDRESS  = UCS_BIT(0), /**< Virtual address */
+    UCP_MEM_ATTR_FIELD_LENGTH   = UCS_BIT(1), /**< The size of memory region */
+    UCP_MEM_ATTR_FIELD_MEM_TYPE = UCS_BIT(2)  /**< Type of allocated memory */
```

Comment:
Is it only allocated memory or also registered memory?
since if a user registers memory, they should already pass the type